vars/publishToNpm.groovy

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/** Library to publish artifacts to NPM registry under @opensearch-project namespace
+@param Map args = [:] args A map of the following parameters
+@param args.repository <required> - Repository to be used to publish the artifact to npm
+@param args.tag <required> - Tag reference to be used to publish the artifact
+*/
+void call(Map args = [:]) {
+
+    checkout([$class: 'GitSCM', branches: [[name: "${args.tag}" ]], userRemoteConfigs: [[url: "${args.repository}" ]]])
+    
+    withCredentials([string(credentialsId: 'publish-to-npm-token', variable: 'NPM_TOKEN')]){
+        sh """npm set registry "https://registry.npmjs.org"; npm set //registry.npmjs.org/:_authToken ${NPM_TOKEN}; npm publish --dry-run && npm publish --access public"""
```

Comment:
Prefer writing this in multiple lines for easier visibility.

```
sh """
    <code>
"""
```

Reply:
Makes its difficult to test things due to whitespace. Hence used the above approach. Similar to https://github.com/opensearch-project/opensearch-build-libraries/blob/main/vars/copyContainer.groovy#L35

Reply:
Huh, I do find it not ideal for reading, what would happen if the code block is too long to put into one line?

Reply:
This was not long enough that's why used this way. If its very long we can use the approach you suggested or try to break it down.

Reply:
I think from my experience, we used to publish the client to staging NPM before final publish. Would that be performed on the Jenkinsfile level which calls this library?

Reply:
@zelinh We won't be dealing with staging. If component team wants to follow that they can do on their own. We will only be directly releasing on prod platform